tests/shellchecks.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Based on https://github.com/koreader/koreader/blob/master/.ci/helper_shellchecks.sh
+
+ANSI_RED="\\033[31;1m"
+ANSI_GREEN="\\033[32;1m"
+ANSI_RESET="\\033[0m"
+
+mapfile -t shellscript_locations < <({ git grep -lE '^#!(/usr)?/bin/(env )?(bash|sh)' && git ls-files ./*.sh; } | sort | uniq)
+
+SHELLSCRIPT_ERROR=0
+
+for shellscript in "${shellscript_locations[@]}"; do
+	echo -e "${ANSI_GREEN}Running shellcheck on ${shellscript}"
+	shellcheck "${shellscript}" || SHELLSCRIPT_ERROR=1
```

Comment:
It could be good to add a note in about this test, and also that it requires installing `shellcheck` and `shfmt`